caravel/assets/javascripts/explorev2/components/AdvancedAnalytics.jsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import SelectArray from './SelectArray';
+import { bindActionCreators } from 'redux';
+import * as actions from '../actions/exploreActions';
+import { connect } from 'react-redux';
+import {
+  rollingOptions, periodRatioTypeOptions, resampleHowOptions,
+  resampleRuleOptions, resampleFillOptions,
+} from '../constants';
+import { formatSelectOptions } from '../../../utils/common';
+
+const propTypes = {
+  actions: React.PropTypes.object.isRequired,
+  rolling: React.PropTypes.string,
+  periodRatioType: React.PropTypes.string,
+  resampleHow: React.PropTypes.string,
+  resampleRule: React.PropTypes.string,
+  resampleFill: React.PropTypes.string,
+};
+
+const defaultProps = {
+  rolling: null,
+  periodRatioType: null,
+  resampleHow: null,
+  resampleRule: null,
+  resampleFill: null,
+};
+
+class AdvancedAnalytics extends React.Component {
+  changeInput(field, event) {
+    this.props.actions.setStateValue(field, event.target.value);
+  }
+  render() {
+    const selects = [
+      {
+        key: 'rolling',
+        title: 'Rolling',
+        options: formatSelectOptions(rollingOptions),
+        value: this.props.rolling,
+      },
+      {
+        key: 'periodRatioType',
+        title: 'Period Ratio Type',
+        options: formatSelectOptions(periodRatioTypeOptions),
+        value: this.props.periodRatioType,
+      },
+      {
+        key: 'resampleHow',
+        title: 'Resample How',
+        options: formatSelectOptions(resampleHowOptions),
+        value: this.props.resampleHow,
+      },
+      {
+        key: 'resampleRule',
+        title: 'Resample Rule',
+        options: formatSelectOptions(resampleRuleOptions),
+        value: this.props.resampleRule,
+      },
+      {
+        key: 'resampleFill',
+        title: 'Resample Fill',
+        options: formatSelectOptions(resampleFillOptions),
+        value: this.props.resampleFill,
+        width: '12',
+      }];
+    return (
+      <div className="panel space-1">
+        <div className="panel-header">Advanced Analytics</div>
```

Comment:
could we create a ControlPanelContainer that takes a header prop and passes in the inputs/controls as children?